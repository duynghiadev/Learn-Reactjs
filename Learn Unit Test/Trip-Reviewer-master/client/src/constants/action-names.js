export const GET_TRIPS = 'get-trips';
export const GET_USERS = 'get-users';
export const ADD_TRIP = 'add-trip';
export const UPDATE_TRIP = 'update-trip';
export const UPDATE_USER = 'update-user';
export const DELETE_TRIP = 'delete-trip';
export const GET_TRIP = 'get-trip';
export const GET_TRIP_ERROR = 'get-trip-error';
export const GET_TRIP_W_REVIEWER = 'get-trip-w-reviewer';
export const GET_TRIP_W_REVIEWER_ERROR = 'get-trip-w-reviewer-error';
export const CLEAR_TRIP_W_REVIEWER = 'clear-trip-w-reviewer';
export const CLEAR_NEW_TRIP = 'clear-new-trip';
export const CLEAR_TRIP = 'clear-trip';
export const USER_REGISTER = 'user-register';
export const USER_LOGIN = 'user-login';
export const RESET_PASSWORD = 'reset-password';
export const USER_AUTH = 'user-auth';
export const GET_USER_REVIEWS = 'get-user-reviews';
