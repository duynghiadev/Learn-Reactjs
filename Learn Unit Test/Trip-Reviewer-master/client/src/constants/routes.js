export const HOME = '/';
export const LOGIN = '/login';
export const RESET = '/reset-password';
export const LOGOUT = '/user/logout';
export const REGISTER = '/user/register';
export const USER_PROFILE = '/user';
export const USER_REVIEWS = '/user/user-reviews';
export const EDIT_PROFILE = '/user/edit-profile';
export const EDIT_POST = '/user/edit-post';
export const ADD_TRIP = '/trip/add';
export const TRIPS = '/trips';
