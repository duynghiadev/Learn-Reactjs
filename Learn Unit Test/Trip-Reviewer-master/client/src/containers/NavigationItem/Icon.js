import styled from 'styled-components';
import FontAwesome from 'react-fontawesome';

const Icon = styled(FontAwesome)`
  color: white;
  width: 20px;
  text-align: center;
`;

export default Icon;
