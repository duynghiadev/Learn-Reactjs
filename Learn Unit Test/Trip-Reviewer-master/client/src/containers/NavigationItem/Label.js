import styled from 'styled-components';
import PropTypes from 'prop-types';

const getCalculatedStyles = ({ sidenav, limit }) => `
  @media (max-width: ${limit}px) {
    margin-left: ${sidenav ? '10px' : '0'}
  }
`;

const Label = styled.span`
  margin-left: 10px;

  &:hover {
    color: white;
  }

  ${props => getCalculatedStyles(props)}
`;

Label.propTypes = {
  limit: PropTypes.number,
  sidenav: PropTypes.number,
};

Label.defaultProps = {
  limit: 0,
  sidenav: 0,
};

export default Label;
