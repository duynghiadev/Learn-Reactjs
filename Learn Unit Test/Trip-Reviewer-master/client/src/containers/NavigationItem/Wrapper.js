import styled from 'styled-components';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const applySidenav = () => `
  color: #bababa;
  padding: 10px;
  border-top: 1px solid #404040;
  line-height: normal;
`;

const applyRegular = limit => `
  color: #ffffff;
  padding: 0 20px;
  display: inline-block;

  &:hover {
    background: #009eb3;
    transform: scale(1.05);
    text-shadow: 1px 1px 20px white;
  }

  &:active {
    background: #0495a8;
    transform: scale(1.01);
    text-shadow: 1px 1px 10px white;
  }

  @media (max-width: ${limit}px) {
    font-size: 16px;
    padding: 0 15px;
  }
`;

const Wrapper = styled(Link)`
  font-weight: 300;
  font-size: 18px;
  display: block;

  ${({ sidenav, limit }) => (sidenav ? applySidenav() : applyRegular(limit))}
`;

Wrapper.propTypes = {
  limit: PropTypes.number,
  sidenav: PropTypes.number,
};

Wrapper.defaultProps = {
  limit: 0,
  sidenav: 0,
};

export default Wrapper;
