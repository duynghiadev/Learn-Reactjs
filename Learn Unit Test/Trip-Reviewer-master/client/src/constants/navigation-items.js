import * as routes from './routes';

export const home = 'HOME';
export const userProfile = 'USER_PROFILE';
export const signup = 'SIGNUP';
export const login = 'LOGIN';
export const userReviews = 'USER_REVIEWS';
export const addTrip = 'ADD_TRIP';
export const logout = 'LOGOUT';

const items = {
  [home]: {
    id: 1,
    icon: 'home',
    text: 'Home',
    link: routes.HOME,
    restricted: false,
  },
  [userProfile]: {
    id: 2,
    icon: 'user',
    text: 'Profile',
    link: routes.USER_PROFILE,
    restricted: true,
  },
  [signup]: {
    id: 3,
    icon: 'user-plus',
    text: 'Signup',
    link: routes.REGISTER,
    restricted: false,
    exclude: true,
  },
  [login]: {
    id: 4,
    icon: 'sign-in-alt',
    text: 'Login',
    link: routes.LOGIN,
    restricted: false,
    exclude: true,
  },
  [userReviews]: {
    id: 5,
    icon: 'edit',
    text: 'Edit',
    link: routes.USER_REVIEWS,
    restricted: true,
  },
  [addTrip]: {
    id: 6,
    icon: 'file-alt',
    text: 'Add',
    link: routes.ADD_TRIP,
    restricted: true,
  },
  [logout]: {
    id: 7,
    icon: 'sign-out-alt',
    text: 'Logout',
    link: routes.LOGOUT,
    restricted: true,
  },
};

export default items;
