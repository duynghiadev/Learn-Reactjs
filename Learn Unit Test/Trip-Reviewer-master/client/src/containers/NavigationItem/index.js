import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import data from '../../constants/navigation-items';

import Wrapper from './Wrapper';
import Label from './Label';
import Icon from './Icon';

const WIDTH_LIMIT = 1024;

const NavigationItem = ({ item, users, sidenav }) => {
  const element = (
    <Wrapper to={data[item].link} sidenav={sidenav ? 1 : 0} limit={WIDTH_LIMIT}>
      <Icon name={data[item].icon} />
      <Label sidenav={sidenav ? 1 : 0} limit={WIDTH_LIMIT}>
        {data[item].text}
      </Label>
    </Wrapper>
  );

  const showItem = () => {
    if (users.login && users.login.isAuth) {
      return !data[item].exclude ? element : null;
    }
    return !data[item].restricted ? element : null;
  };

  return <React.Fragment>{showItem()}</React.Fragment>;
};

NavigationItem.propTypes = {
  users: PropTypes.object.isRequired,
  item: PropTypes.string.isRequired,
  sidenav: PropTypes.bool,
};

NavigationItem.defaultProps = {
  sidenav: false,
};

function mapStateToProps(state) {
  return {
    users: state.users,
  };
}

export default connect(mapStateToProps)(NavigationItem);
