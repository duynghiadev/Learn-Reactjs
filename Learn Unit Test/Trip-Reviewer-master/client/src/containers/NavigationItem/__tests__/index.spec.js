import React from 'react';
import configureStore from 'redux-mock-store';
import { create } from 'react-test-renderer';

import NavigationItem from '..';
import { home, userReviews, signup } from '../../../constants/navigation-items';

jest.mock('../Wrapper', () => 'Wrapper');
jest.mock('../Label', () => 'Label');
jest.mock('../Icon', () => 'Icon');

const mockStore = configureStore();

const mockComponent = (initialState = {}, props) => {
  const store = mockStore(initialState);

  return <NavigationItem {...props} store={store} />;
};

describe('<NavigationItem />', () => {
  it('should render component when no authenticated', () => {
    const initialState = {
      users: {
        login: {
          isAuth: false,
        },
      },
    };
    const props = {
      item: home,
    };
    const tree = create(mockComponent(initialState, props)).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render component when no authenticated and restricted', () => {
    const initialState = {
      users: {
        login: {
          isAuth: false,
        },
      },
    };
    const props = {
      item: userReviews,
    };
    const tree = create(mockComponent(initialState, props)).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render component with authentication', () => {
    const initialState = {
      users: {
        login: {
          isAuth: true,
        },
      },
    };
    const props = {
      item: home,
    };
    const tree = create(mockComponent(initialState, props)).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render component with authentication and excluded', () => {
    const initialState = {
      users: {
        login: {
          isAuth: true,
        },
      },
    };
    const props = {
      item: signup,
    };
    const tree = create(mockComponent(initialState, props)).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render component with authentication and sidenav', () => {
    const initialState = {
      users: {
        login: {
          isAuth: true,
        },
      },
    };
    const props = {
      item: home,
      sidenav: true,
    };
    const tree = create(mockComponent(initialState, props)).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
