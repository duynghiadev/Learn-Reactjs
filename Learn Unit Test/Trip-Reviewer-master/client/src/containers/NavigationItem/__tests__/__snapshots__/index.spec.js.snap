// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationItem /> should render component when no authenticated 1`] = `
<Wrapper
  limit={1024}
  sidenav={0}
  to="/"
>
  <Icon
    name="home"
  />
  <Label
    limit={1024}
    sidenav={0}
  >
    Home
  </Label>
</Wrapper>
`;

exports[`<NavigationItem /> should render component when no authenticated and restricted 1`] = `null`;

exports[`<NavigationItem /> should render component with authentication 1`] = `
<Wrapper
  limit={1024}
  sidenav={0}
  to="/"
>
  <Icon
    name="home"
  />
  <Label
    limit={1024}
    sidenav={0}
  >
    Home
  </Label>
</Wrapper>
`;

exports[`<NavigationItem /> should render component with authentication and excluded 1`] = `null`;

exports[`<NavigationItem /> should render component with authentication and sidenav 1`] = `
<Wrapper
  limit={1024}
  sidenav={1}
  to="/"
>
  <Icon
    name="home"
  />
  <Label
    limit={1024}
    sidenav={1}
  >
    Home
  </Label>
</Wrapper>
`;
